src/screens/TopicPageScreen/index.js

```diff
@@ -162,7 +162,7 @@ const TopicPageScreen = (props) => {
     try {
       const resultTopicDetail = await getTopics(domain);
       if (resultTopicDetail.data) {
-        const detail = resultTopicDetail.data[0];
+        const detail = resultTopicDetail.data.find((item) => item.name === domain) || {};
         setTopicDetail(detail);
         handleFollowData(detail.is_followed_by);
         setMemberCount(Number(detail.followersCount));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code is more robust as it searches for the topic detail by name instead of assuming it's the first item in the data array. However, there's a potential issue here: if `item.name === domain` doesn't find a match, it will default to an empty object `{}`. This could lead to undefined errors when trying to access properties like `detail.is_followed_by` and `detail.followersCount`. 

To handle this, you should add checks before accessing these properties.

```diff
-        const detail = resultTopicDetail.data.find((item) => item.name === domain) || {};
+        const detail = resultTopicDetail.data.find((item) => item.name === domain);
+        if (!detail) {
+            throw new Error(`No topic found with domain: ${domain}`);
+        }
         setTopicDetail(detail);
-        handleFollowData(detail.is_followed_by);
-        setMemberCount(Number(detail.followersCount));
+        if (detail.is_followed_by !== undefined) {
+            handleFollowData(detail.is_followed_by);
+        }
+        if (detail.followersCount !== undefined) {
+            setMemberCount(Number(detail.followersCount));
+        }
```
This way, we ensure that we only try to access properties on `detail` if it exists and has the expected structure. If not, we throw an error which can be caught and handled appropriately.


<!-- This is an auto-generated comment by OSS CodeRabbit -->